hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/converter/FSConfigConverterTestCommons.java

```diff
@@ -58,9 +58,32 @@ public FSConfigConverterTestCommons() {
 
   public void setUp() throws IOException {
     File d = new File(TEST_DIR, "conversion-output");
+
     if (d.exists()) {
-      FileUtils.deleteDirectory(d);
+      // Retry mechanism to ensure file handles are released
+      int maxRetries = 3;
+      int attempts = 0;
+      boolean deleted = false;
+
+      while (attempts < maxRetries && !deleted) {
+        try {
+          FileUtils.deleteDirectory(d);
+          deleted = true;
+        } catch (IOException e) {
+          attempts++;
+          System.gc(); // Hint JVM to run GC to close any unreferenced file streams
+          try {
+            Thread.sleep(100); // Wait a bit before retrying
+          } catch (InterruptedException ie) {
+            Thread.currentThread().interrupt();
+          }
+          if (attempts == maxRetries) {
+            throw new IOException("Failed to delete directory after retries: " + d.getAbsolutePath(), e);
+          }
+        }
+      }
```

Comment:
This still may not guarantee the folder deletion. A better way to solve this would be to configure this class to never run the unit tests in parallel. You can use the `@Execution(ExecutionMode.SAME_THREAD)` annotation on the class in JUnit 5.

Could you please use an equivalent approach for the version of JUnit used here?